File: mobile/android/src/main/kotlin/com/otravelz/android/data/network/dto/CoreDtos.kt
package com.otravelz.android.data.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class HealthResponseDto(
    val status: String,
    val version: String,
    @SerialName("git_sha") val gitSha: String? = null,
    @SerialName("alembic_version") val alembicVersion: String? = null,
    val database: String? = null
)

@Serializable
data class ReadyResponseDto(
    val status: String,
    val database: String? = null
)

@Serializable
data class ApiErrorDetailDto(
    val code: String,
    val message: String,
    val field: String? = null
)

@Serializable
data class ApiErrorResponseDto(
    val error: ApiErrorDetailDto,
    val details: List<Map<String, String?>> = emptyList()
)

File: mobile/ios/OTravelz/Networking/DTO/CoreDTOs.swift
import Foundation

public struct HealthResponseDTO: Codable, Sendable {
    public let status: String
    public let version: String
    public let gitSha: String?
    public let alembicVersion: String?
    public let database: String?

    enum CodingKeys: String, CodingKey {
        case status
        case version
        case gitSha = "git_sha"
        case alembicVersion = "alembic_version"
        case database
    }
}

public struct ReadyResponseDTO: Codable, Sendable {
    public let status: String
    public let database: String?
}

public struct APIErrorDetailDTO: Codable, Sendable {
    public let code: String
    public let message: String
    public let field: String?
}

public struct APIErrorResponseDTO: Codable, Sendable {
    public let error: APIErrorDetailDTO
}
